src/plugins/citation.jl

```diff
@@ -0,0 +1,43 @@
+"""
+    Citation(; readme_section::Bool=false)
+
+Add `Citation` to a template's plugins to add a `CITATION.bib` file to
+generated repositories, and an appropriate section in the README.
+
+# Keyword Arguments:
+* `readme_section::Bool=false`: whether to add a section in the readme pointing to `CITATION.bib`.
+"""
+struct Citation <: GenericPlugin
+    gitignore::Vector{AbstractString}
+    src::Union{String, Nothing}
+    dest::AbstractString
+    badges::Vector{Badge}
+    view::Dict{String, Any}
+    readme_section::Bool
+    function Citation(; readme_section::Bool=false)
+        new(
+            [],
+            nothing,
+            "CITATION.bib",
+            [],
+            Dict{String, Any}(),
+            readme_section,
+        )
+    end
+end
+
+interactive(::Type{Citation}) = interactive(Citation; readme_section=false)
+
+function gen_plugin(p::Citation, t::Template, pkg_name::AbstractString)
+    pkg_dir = joinpath(t.dir, pkg_name)
+    text = """@misc{$pkg_name.jl,
+              \tauthor  = {{$(t.author)}},\n
+              \ttitle   = {{$(pkg_name).jl}},\n
```

Comment:
```suggestion
              \ttitle   = {{$pkg_name.jl}},\n
```

Also, same question as `author`.

Reply:
I find it easier on the eyes to have explicit parentheses when the variable name is long-ish, otherwise it can lead to ambiguities on what belongs to the interpolated expression: $(pkg_name).jl or $(pkg_name.jl)